tensorflow/python/ops/array_grad.py

```diff
@@ -831,12 +831,9 @@ def _QuantizeAndDequantizeV3Grad(_, grad):
 
 @ops.RegisterGradient("ExtractImagePatches")
 def _ExtractImagePatchesGrad(op, grad):
-  batch_size, rows_in, cols_in, channels = [
-      dim.value for dim in op.inputs[0].shape.dims
-  ]
-  input_bhwc = array_ops.shape(op.inputs[0])
-  batch_size = input_bhwc[0]
-  channels = input_bhwc[3]
+  input_bhwc = math_ops.cast(array_ops.shape(op.inputs[0]), ops.dtypes.int64)
```

Comment:
Use array_ops.shape(op.inputs[0], out_type=dtypes.int64) instead of an explicit cast.

Also don't use `ops.dtypes`, just import the dtypes module here (we don't want to depend on what modules are imported by another module).

Reply:
Thank you for your comment. I corrected the code as instructed.